ports/stm32/uart.c

```diff
@@ -536,6 +541,15 @@ void uart_irq_handler(mp_uint_t uart_id) {
             }
         }
     }
+    if (__HAL_UART_GET_FLAG(&self->uart, UART_FLAG_IDLE) != RESET) {
+        __HAL_UART_CLEAR_IDLEFLAG(&self->uart);
+        self->irq.flags |= IRQ_FLAG_RX_IDLE;
+    }
+    self->irq.flags &= self->irq.enable;
+    if (self->irq.flags != IRQ_FLAG_NONE) {
```

Comment:
To simplify this could it could use native ST SR flags, something like this (`SR` for F4, `ISR` for L4):
```
self->irq.flags = self->uart.Instance->SR & self->irq.enable;
if (self->irq.flags) {
```
That would require self->irq.enable to also have bits set in terms of the native ST SR bit definitions, eg `RX_IDLE=1<<4`.

Reply:
If the flags are decoupled from the native representation there is a possibility to generalize the IRQ flags for all platforms (which was my aim). However IRQs are always close to the machine...

Reply:
There could still be a lot of code generalised to all platforms (which is a good idea BTW!).  The flags entry can just be a uint16_t and then it's up to the platform to provide the values/constants.